tools/releasing/create_release_branch.sh

```diff
@@ -59,6 +59,22 @@ git checkout -b $target_branch
 #change version in all pom files
 find . -name 'pom.xml' -type f -exec perl -pi -e 's#<version>(.*)'$OLD_VERSION'(.*)</version>#<version>${1}'$NEW_VERSION'${2}</version>#' {} \;
 
+ultra=$(echo "${NEW_VERSION}" | perl -p -e 's#(\d+)\.\d+\.\d+#$1#')
+major=$(echo "${NEW_VERSION}" | perl -p -e 's#\d+\.(\d+)\.\d+#$1#')
+minor=$(echo "${NEW_VERSION}" | perl -p -e 's#\d+\.\d+\.(\d+)#$1#')
+
+# update japicmp configuration
+if [[ ${minor} == "0" ]]; then
+  # set japicmp reference version to previous major version
+  perl -pi -e 's#(<japicmp.referenceVersion>).*(</japicmp.referenceVersion>)#${1}'${ultra}.$((${major}-1)).0'${2}#' pom.xml
+else
+  # set japicmp reference version to previous minor version
+  perl -pi -e 's#(<japicmp.referenceVersion>).*(</japicmp.referenceVersion>)#${1}'${ultra}.${major}.$(($minor - 1))'${2}#' pom.xml
+  # harden compatibility constraints for minor versions
+  perl -pi -e 's#<!--(<include>\@org.apache.flink.annotation.PublicEvolving</include>)-->#${1}#' pom.xml
+  perl -pi -e 's#\t+<exclude>\@org.apache.flink.annotation.PublicEvolving.*\n##' pom.xml
+fi
```

Comment:
Shouldn't these changes go into `releasing/update_branch_version.sh`? This file here seems to be used for preparing release candidate branches. (According to https://cwiki.apache.org/confluence/display/FLINK/Creating+a+Flink+Release#CreatingaFlinkRelease-Createareleasebranch)

Reply:
hmm, you're right, but I think we should have both.

For the release-X.Y branches we have to update the configuration manually, so this PR is more of a final safeguard for actual releases in case the the update is forgotten, which is basically guaranteed to happen at some point.

Reply:
Ah, so the idea is that the build of a `release-x.y.z-rci` branch would fail (as a last resort) because the japicmp plugin gets reconfigured as part of the release process.
But ideally the release managers manually uncomment this one line in the pom in the `release-X.Y` branch.

Reply:
It wasn't the original idea, I just remembered wrong how the release process works.

As for the manual work, the RM has to un-comment the line, remove the exclusion and update the reference version to the last release minor version.

Ultimately the logic is identical, so I'll see if I can refactor this update into a separate script that the RM can call.